cryptoki/src/session/object_management.rs

```diff
@@ -14,6 +14,42 @@ const MAX_OBJECT_COUNT: usize = 10;
 
 impl Session {
     /// Search for session objects matching a template
+    ///
+    /// # Arguments
+    /// * `template` - A [Attribute] of search parameters that will be used
+    ///                 to find objects.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use cryptoki::session::Session;
+    /// use cryptoki::context::Pkcs11;
+    /// use cryptoki::object::{Attribute, AttributeType, CertificateType, ObjectClass, ObjectHandle};
+    ///
+    /// let mut client = Pkcs11::new(
+    ///     std::env::var("PKCS11_SOFTHSM2_MODULE")
+    ///        .unwrap_or_else(|_| "/usr/local/lib/softhsm/libsofthsm2.so".to_string()),
+    /// ).expect("Failed to load library");
+    /// client.initialize(cryptoki::context::CInitializeArgs::OsThreads).expect("Failed to init session");
+    ///
+    /// // Use the first slot
+    /// let slots = client.get_all_slots().expect("Failed to list slots on card");
+    /// let slot = slots.first().expect("Failed to find 1st slot");
+    /// let session = client.open_ro_session(*slot).expect("Failed to open read-only session");
+    ///
+    /// // Get handles to all of the x509 certificates on the card
+    /// let search = vec![Attribute::Class(ObjectClass::CERTIFICATE), Attribute::CertificateType(CertificateType::X_509)];
+    /// for handle in session.find_objects(&search).expect("Failed to get certificates from card") {
+    ///   // each cert: get the "value" which will be the raw certificate data
+    ///   for value in session.get_attributes(handle, &[AttributeType::Value]).expect("Failed to search for object atrributes") {
+    ///      match value {
```

Comment:
I think this match with one arm can be more succinctly expressed using something like: `if let Attribute::Value(value) = value { ...`